eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1913,66 +1913,70 @@ public function deleteTranslation(ContentInfo $contentInfo, $languageCode)
                 'Specified translation is the main translation of the Content Object'
             );
         }
-        // before actual removal, collect information on Versions
-        $versions = [];
-        $singleLangVersions = [];
-        foreach ($this->loadVersions($contentInfo) as $versionInfo) {
-            // check if user is authorized to delete Version
-            if (!$this->repository->canUser('content', 'remove', $versionInfo)) {
-                throw new UnauthorizedException(
-                    'content',
-                    'remove',
-                    [
-                        'contentId' => $contentInfo->id,
-                        'versionNo' => $versionInfo->versionNo,
-                    ]
-                );
-            }
-            // check if the specified translation exists for the Version
-            if (!in_array($languageCode, $versionInfo->languageCodes)) {
-                // if translation does not exist, simply ignore Version (see InvalidArgumentException later on)
-                continue;
-            }
-            // check if the specified translation is the only one
-            if (count($versionInfo->languageCodes) < 2) {
-                $singleLangVersions[] = $versionInfo->versionNo;
-            } else {
-                // otherwise add Version to the list of valid Versions to be processed
-                $versions[] = $versionInfo->versionNo;
+
+        $translationWasFound = false;
+        $this->repository->beginTransaction();
+        try {
+            foreach ($this->loadVersions($contentInfo) as $versionInfo) {
+                // check if user is authorized to delete Version
```

Comment:
Nitpick: those comments should be removed, imho (not a blocker). They paraphrase what the code says. I welcome the willingness to document what is done, but the best documentation remains readable code.

Reply:
True. Removed and amended via ef05fdc.